tokio-postgres/src/connect_raw.rs

```diff
@@ -141,8 +141,13 @@ where
     T: AsyncRead + AsyncWrite + Unpin,
 {
     match stream.try_next().await.map_err(Error::io)? {
-        Some(Message::AuthenticationOk) => return Ok(()),
+        Some(Message::AuthenticationOk) => {
+            no_channel_binding(config)?;
+            return Ok(());
+        }
         Some(Message::AuthenticationCleartextPassword) => {
+            no_channel_binding(config)?;
```

Comment:
It feels a little fragile to try to check this for each auth method that might leak information to a malicious server. Perhaps check whether the auth request is expected, given the config, at the top so it covers everything?

Reply:
It could be handled in a separate match, but I'd be worried about forgetting to update both locations if we add support for a new authentication method.